### ISSUE TITLE
Procurement does not consider the sequence and the line choice

### ISSUE THREAD

**Issue by ybofr** [odoo/odoo#5835]
I have tested this on runbot v8.

I have a product with 2 possible routes: make2order and buy. I have set the make2order route at sequence 20 and the buy at sequence 10.

When i create a sales order, it does not matter what I put in the route, the make2order is always selected. So, I choose the route Buy, but the system forces a make2order.

First, I was expecting to have the first in the sequence chosen automatically, second, I do not understand why the system is overruling me, making the choice effectively impossible.

And in the contrary to the to fast closed issue, I need to add the following:

I want to be able to choose if I purchase specifically for a customer or if I purchase to stock. This has NOTHING to do with dropship. In the dropship process, the delivery is done from the supplier to the customer. In the make2order, I receive the goods in stock and deliver it myself. So, I NEED to be able to choose.

And by the way, it was working perfectly in V7.

Thanks for any correction

Eva P:

**Comment by spadae22** [comment]
Yes I agree. v8 did fix many issues and the WMS is much improved, but along the way simple issue as stated above and reorder rule for Buy, if not available does not show an exception. I am training two purchase managers with 3000+ products and we have to look at negative stock every day to see what did not get procured, it is running, but that is about it, running knowwhere

**Comment by falinwa** [comment]
yes, I confirmed this.
and it must be supported for multi-company too,
the routes has a company_id, so it must respected that configuration too.
Right now I think its not respected that yet.

**Comment by jco-odoo** [comment]
A product has a orderpoint_ids field (it is called Minimum Stock Rules), so how much easier is it to check the products with negative stock and no orderpoints.  In v7, even if a product has an orderpoint it will go into exception when there is not enough stock available, while you already get this message on confirmation of the sale order.  

@ybofr  If you create a product without checking MTO on the routes and when creating a sale order you only put MTO in the line when you need it, this will work.

**Comment by ybofr** [comment]
I do not want to use the Minimum stock rule. I know exactly what I want from the procurement point of view. 
@jco-odoo Thanks. So, the system was changed since V7.0. On previous version, I had to allow the route by selecting it on the product. Now, it is not needed or better said, it is forbidden, if you want to keep the choice. I believe that the meaning of route on the product form is then a little bit unclear or useless.

So, my issue is solved.
